yarn-project/prover-client/src/prover-agent/memory-proving-queue.ts

```diff
@@ -31,34 +31,35 @@ import type {
 } from '@aztec/circuits.js';
 import { randomBytes } from '@aztec/foundation/crypto';
 import { AbortError, TimeoutError } from '@aztec/foundation/error';
-import { MemoryFifo } from '@aztec/foundation/fifo';
 import { createDebugLogger } from '@aztec/foundation/log';
 import { type PromiseWithResolvers, RunningPromise, promiseWithResolvers } from '@aztec/foundation/promise';
+import { PriorityMemoryQueue } from '@aztec/foundation/queue';
 import { serializeToBuffer } from '@aztec/foundation/serialize';
 import { type TelemetryClient } from '@aztec/telemetry-client';
 
 import { ProvingQueueMetrics } from './queue_metrics.js';
 
-type ProvingJobWithResolvers<T extends ProvingRequest = ProvingRequest> = {
-  id: string;
-  request: T;
-  signal?: AbortSignal;
-  attempts: number;
-  heartbeat: number;
-} & PromiseWithResolvers<ProvingRequestResult<T['type']>>;
+type ProvingJobWithResolvers<T extends ProvingRequest = ProvingRequest> = ProvingJob<T> &
+  PromiseWithResolvers<ProvingRequestResult<T['type']>> & {
+    signal?: AbortSignal;
+    epochNumber?: number;
```

Comment:
Would it be possible to make `epochNumber` mandatory?

Reply:
Makes sense. I'm deferring this though until we decide if we want to stick to epoch numbers or use something different for priorities (see #7925).